boa/src/object/mod.rs

```diff
@@ -401,6 +401,14 @@ impl Object {
         self.prototype = prototype
     }
 
+    /// Similar to `Value::new_object`, but you can pass a prototype to create from, plus a kind
+    pub fn new_object_from_prototype(proto: Value, data: ObjectData) -> GcObject {
```

Comment:
```suggestion
    #[inline]
    pub fn new_object_from_prototype(proto: Value, data: ObjectData) -> GcObject {
```
we should add inline so it can be inlined in other crates